### PR TITLE
Add Cvode Documentation

### PR DESCRIPTION
**Pull request by AnHeuermann**
- Also fixed wrong order in tooltip

### CODE REVIEW

File: doc/UsersGuide/source/solving.rst [hunk 1]
```diff
@@ -73,6 +73,29 @@ solver and furthermore it has the following IDA specific flags:
...
+
+The CVODE solver is part of sundials: SUite of Nonlinear and
+DIfferential/ALgebraic equation Solvers :cite:`Hindmarsh:2005`.
+CVODE solves initial value problems for ordinary differential equation (ODE)
+systems with variable-order, variable-step multistep methods.
+
+In OpenModelica CVODE uses a combination of Backward Differentiation
+Formulas (BDFs), with order varying between 1 and 5, and a Modified Newton
+iteration with fixed Jacobian as default.
+This setting is advised for stiff problems and therefore chooses by default.
```
Comment (sjoelund) [inline doc/UsersGuide/source/solving.rst:90]: Chosen by default?

File: doc/UsersGuide/source/solving.rst [hunk 2]
```diff
@@ -73,6 +73,29 @@ solver and furthermore it has the following IDA specific flags:
...
+
+CVODE
+~~~~~
+
+The CVODE solver is part of sundials: SUite of Nonlinear and
+DIfferential/ALgebraic equation Solvers :cite:`Hindmarsh:2005`.
+CVODE solves initial value problems for ordinary differential equation (ODE)
+systems with variable-order, variable-step multistep methods.
+
+In OpenModelica CVODE uses a combination of Backward Differentiation
```
Comment (sjoelund) [inline doc/UsersGuide/source/solving.rst:87]: In OpenModelica, CVODE...

File: doc/UsersGuide/source/solving.rst [hunk 3]
```diff
@@ -73,6 +73,29 @@ solver and furthermore it has the following IDA specific flags:
...
+Formulas (BDFs), with order varying between 1 and 5, and a Modified Newton
```
Comment (sjoelund) [inline doc/UsersGuide/source/solving.rst:88]: modified Newton

File: doc/UsersGuide/source/solving.rst [hunk 4]
```diff
@@ -73,6 +73,29 @@ solver and furthermore it has the following IDA specific flags:
...
+iteration with fixed Jacobian as default.
+This setting is advised for stiff problems and therefore chooses by default.
+For non-stiff problems a combination of Adams-Moulton formulas, with the order
+varying between 1 and 12, and functional iteration as non-linear solver can
```
Comment (sjoelund) [inline doc/UsersGuide/source/solving.rst:92]: Not sure exactly what you mean here

File: doc/UsersGuide/source/solving.rst (hunk 1 above)
Comment (AnHeuermann) [reply doc/UsersGuide/source/solving.rst:90]: If you just use `-s=cvode` it will use BDF as linear multistep method and Newton as non-linear solver.
May be I should reword it to: `This setting is advised for stiff problems and enabled per default.`

File: doc/UsersGuide/source/solving.rst (hunk 4 above)
Comment (AnHeuermann) [reply doc/UsersGuide/source/solving.rst:92]:
````
For non-stiff problems an Adams-Moulton formula (varying order 1 to 12) is used as linear multi-step method
together with a fixed-point iteration as non-linear solver method.
```
Better?

Comment (sjoelund) [reply doc/UsersGuide/source/solving.rst]: It's a nonlinear solver internal to cvode that is not provided by us, right?

File: doc/UsersGuide/source/solving.rst (hunk 1 above)
Comment (sjoelund) [reply doc/UsersGuide/source/solving.rst]: If it's the OpenModelica nonlinear solver (unrelated to cvode internals), I guess that should be clarified as well.

File: doc/UsersGuide/source/solving.rst (hunk 4 above)
Comment (AnHeuermann) [reply doc/UsersGuide/source/solving.rst]: Correct

File: doc/UsersGuide/source/solving.rst (hunk 1 above)
Comment (AnHeuermann) [reply doc/UsersGuide/source/solving.rst]: I can add that as well

File: doc/UsersGuide/source/solving.rst [hunk 5]
```diff
@@ -73,6 +73,35 @@ solver and furthermore it has the following IDA specific flags:
...
+
+The CVODE solver is part of sundials: SUite of Nonlinear and
+DIfferential/ALgebraic equation Solvers :cite:`Hindmarsh:2005`.
+CVODE solves initial value problems for ordinary differential equation (ODE)
+systems with variable-order, variable-step multistep methods.
+
+In OpenModelica, CVODE uses a combination of Backward Differentiation
+Formulas (varying order 1 to 5) as linear multi-step method and a modified
+Newton iteration with fixed Jacobian as non-linear solver per default.
+This setting is advised for stiff problems which a very common for Modelica
```
Comment (adeas31) [inline doc/UsersGuide/source/solving.rst:90]: Shouldn't it be `are` instead of `a` here.

`This setting is advised for stiff problems which are very common for Modelica models.`

Comment (AnHeuermann) [reply doc/UsersGuide/source/solving.rst:90]: Yes

**Comment by casella** [comment]
Excellent, thanks!
